tests/datasets/test_upload/test_dataset_upload.py

```diff
@@ -0,0 +1,368 @@
+
+import csv
+import pytest
+import pandas as pd
+from io import StringIO, BytesIO
+from django.core.exceptions import ValidationError
+from django.core.files.uploadedfile import SimpleUploadedFile
+
+from tests.datasets.factories import (
+    GeographyFactory, GeographyHierarchyFactory, DatasetFactory,
+    DatasetFileFactory, IndicatorFactory, GroupFactory, DatasetDataFactory
+)
+from tests.profile.factories import ProfileFactory
+from wazimap_ng.datasets.models import DatasetData, Group
+from wazimap_ng.datasets.tasks import process_uploaded_file, indicator_data_extraction
+
+from .upload_data import (
+    dataset_upload_data, default_result_set, required_headers
+)
+
+
+@pytest.mark.django_db
+class TestBaseDatasetUpload:
+    @pytest.fixture(autouse=True)
+    def setUp(self):
+        self.ZA_geo = GeographyFactory(code="ZA")
```

Comment:
Wouldn't it be better to use fixtures here? It is more idiomatic to pytest - and removes boilerplate

Reply:
Checking how to use fixtures in pytests in your tests and docs @adieyal 

Also In initial upload we are not checking for duplicate rows of data and there can duplicate set of data with different counts as I have done in tests. Should we fix that in this PR or open a new task for that ?

Reply:
Done 👍